scope_extensions-explainer.md

```diff
@@ -1,126 +1,216 @@
-associated origins.
+To extend app scope, a developer will:
 
+1. Add a `scope_extensions` section to the web app manifest which lists one or
+  more extensions. 
 
-## Proposal
-To extend the app scope, a developer must modify the web app manifest and host
-one or more association files.
+1. Host a `.well-known/web-app-origin-association` file at each additional
+  origin. This file establishes a two-way handshake between a unique app and the
+  origin owner. It also contains any fine-grain URL filters needed to control
+  scope.
 
-### Web app manifest
-Add a `scope_extensions` member to the web app manifest specifying a list of
-origins to include in the extended app scope. 
+### Manifest
 
 Example manifest located at `https://example.com/manifest.webmanifest`:
    ```json
    {
-     "id": "/",
-     "name": "Example App",
+     "id": "https://example.com/app",
+     "name": "My App",
      "display": "standalone",
      "start_url": "/app/index.html",
      "scope": "/app",
      "scope_extensions": [
-       { "type": "registrable_domain", "value": "https://example.com" },
-       { "type": "registrable_domain", "value": "https://example.co.uk" },
-       { "type": "origin", "value": "https://helpcenter.example-help-center.com" }
+       { "type": "origin", "value": "https://example.co.uk" },
+       { "type": "origin", "value": "https://help.example.com" }
      ]
    }
    ```
-In this example the "Example" app has a regular scope of
-`http://example.com/app` and is extending its app scope to the registrable
-domains of `https://example.com` and `https://example.co.uk`, as well as the
-single origin `https://helpcenter.example-help-center.com`. 
-
-Each object in `scope_extension` must contain both `type` and `value` string
-fields. The value of `type` must be one of `["origin" | "registrable_domain"]`.
-The `value` field must contain a valid URL. 
+The "Example" app has a regular scope of `http://example.com/app` and is
+extending its app scope to the origins `https://example.co.uk` and
+`https://help.example.com`.
 
-| type   | Behavior |
-|--------|----------|
-| `origin` | The URL in `value` is converted to an [origin](https://html.spec.whatwg.org/multipage/browsers.html#concept-origin-tuple). The `/` path of that origin is added to the extended scope.|
-| `registrable_domain` | The URL in `value` is converted to a [registrable domain](https://url.spec.whatwg.org/#host-registrable-domain). The `/` paths of all sub-domain origins within the registrable domain are added to the extended scope.|
-
-This format allows for both backward and forward compatibility. For example, a
-new type of entry could filter the paths of an origin added to the extended
-scope. Entries with types are that not recognized by a user agent should be
-ignored.
+* Each entry in `scope_extensions` must contain both `type` and `value` string
+fields.
+* `type` must be `"origin"`. Other types could be added in the future.
```

Comment:
May be worth mentioning that "future" would be related to the non-example at line 57

Reply:
Done.